podcasts/Episode Info Coordinator/ShowInfoCoordinator.swift

```diff
@@ -88,13 +88,15 @@ actor ShowInfoCoordinator: ShowInfoCoordinating {
             return try await task.value
         }
 
-        let task = Task<Episode.Metadata?, Error> { [unowned self] in
+        let task = Task<Episode.Metadata?, Error> { [weak self] in
+            guard let self else { return nil }
+
             do {
                 let data = try await dataRetriever.loadEpisodeDataFromCache(for: podcastUuid, episodeUuid: episodeUuid)
-                requestingShowInfo[episodeUuid] = nil
+                await setRequestingShowInfoToNil(for: episodeUuid)
```

Comment:
Nit: Replace this call and the one inside the catch for a `defer` call before the do statement?

Reply:
We can't, unfortunately :(

<img width="1187" alt="Screenshot 2024-08-21 at 11 22 43" src="https://github.com/user-attachments/assets/94e51e98-20be-4cae-aa39-f66481c89f0d">

Reply:
TIL